spark3/src/main/java/org/apache/iceberg/spark/procedures/BaseProcedure.java

```diff
@@ -75,25 +71,15 @@ protected BaseProcedure(TableCatalog tableCatalog) {
   }
 
   // we have to parse both namespace and name as they may be quoted
-  protected Identifier toIdentifier(String namespaceAsString, String name) {
-    String[] namespaceParts = parseMultipartIdentifier(namespaceAsString);
-
-    String[] nameParts = parseMultipartIdentifier(name);
-    Preconditions.checkArgument(nameParts.length == 1, "Name must consist of one part: %s", name);
-
-    return Identifier.of(namespaceParts, nameParts[0]);
-  }
-
-  private String[] parseMultipartIdentifier(String identifierAsString) {
+  protected Identifier toIdentifier(String identifier) {
+    Spark3Util.CatalogAndIdentifier catalogAndIdentifier;
```

Comment:
Shall we do an import so that we can refer to `CatalogAndIdentifier` directly?

Reply:
Can do